clar2wasm/src/words/functions.rs

```diff
@@ -211,4 +236,74 @@ mod tests {
             evaluate("(list 'S33GG8QRVWKM7AR8EFN0KZHWD5ZXPHKCWPCZ07BHE.A 'S530MSMK2C8KCDN61ZFMYKFXBHKAP6P32P4S74CJ3.a)")
         );
     }
+
+    #[test]
+    fn validate_define_private() {
+        // Reserved keyword
+        crosscheck("(define-private (map) (ok true))", Err(()));
+        // Custom function name
+        crosscheck("(define-private (a) (ok true))", Ok(None));
+        // Custom functiona name duplicate
+        crosscheck(
+            "(define-private (a) (ok true))(define-private (a) (ok true))",
+            Err(()),
+        );
+    }
+
+    #[test]
+    fn validate_define_public() {
+        // Reserved keyword
+        crosscheck("(define-public (map) (ok true))", Err(()));
+        // Custom function name
+        crosscheck("(define-public (a) (ok true))", Ok(None));
+        // Custom functiona name duplicate
+        crosscheck(
+            "(define-public (a) (ok true))(define-public (a) (ok true))",
+            Err(()),
+        );
+    }
+
+    #[test]
+    fn validate_define_read_only() {
+        // Rserved keyword
+        crosscheck("(define-read-only (map) (ok true))", Err(()));
+        // Custom function name
+        crosscheck("(define-read-only (a) (ok true))", Ok(None));
+        // Custom function name duplicate
+        crosscheck(
+            "(define-read-only (a) (ok true))(define-read-only (a) (ok true))",
+            Err(()),
+        );
+    }
+
+    #[test]
+    fn validate_define_x_epochs() {
```

Comment:
`define_x` ?

Reply:
It had two different methods, that's why. I should split them

Reply:
Done